synIndex.R

```diff
@@ -25,44 +34,119 @@ source('params.R')
 # Get bucket params and file list
 #############
 ## Get a list of all Objects in the PRE_ETL S3 bucket 
-s3lsBucketObjects(source_bucket = paste0('s3://', PRE_ETL_BUCKET,'/'),
-                  output_file = FILE_LIST_OUTPUT)
+# s3lsBucketObjects(source_bucket = paste0('s3://', PRE_ETL_BUCKET,'/'),
+#                   output_file = FILE_LIST_OUTPUT)
 
 # Get bucket params
 bucket_params <- list(uploadType='S3',
-                    concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
-                    bucket=SOURCE_BUCKET)
+                      concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
+                      bucket=PRE_ETL_BUCKET)
 
 # The above list is just to verify/reference, 
 # since we will replicate the structure locally
 # we will work with that
 
-## Get a list of all local files
-localFileList <- list.files(path = AWS_DOWNLOAD_LOCATION,
-                            all.files = TRUE, # get hidden files too
-                            recursive = TRUE, # get files inside sub-folders (if any)
-                            full.names = FALSE) # to get the directory path prepended to the file name
+# all folders inside the AWS_DOWNLOAD_LOCATION
+# When we create dataFileHandleId in synapse it will create some folders in the source S3 bucket
+# we don't need those
+localDirs <- list.dirs(path = AWS_DOWNLOAD_LOCATION, full.names = FALSE) 
+
+## The folders we want to show in synapse
+## This would depend on the how the data is organized in the main S3 INGRESS bucket
+FOLDERS_TO_SYNC_SYNAPSE <- c('adults',
+                             'pregnant',
+                             'pediatric') 
+
+## From the local AWS location remove all folders that are not the ones selected above
+dirs_to_delete <- dplyr::setdiff(localDirs, c(FOLDERS_TO_SYNC_SYNAPSE, "")) # "" is the local directory, a result of list.dirs(.. full.names=FALSE)
+
+## delete the unwanted directories before creating a upload manifest
+for(dir_ in dirs_to_delete){
+  unlink(paste0(AWS_DOWNLOAD_LOCATION,"/",dir_), recursive = TRUE)
+}
+
```

Comment:
What about logging these extra files.  Why can we assume that these are not relevant.

Reply:
Because we only need files in the folders CE drops them in. These extra files might be generated because of indexing (just dataPointers).